content/en/docs/reference/using-api/cel.md

```diff
@@ -0,0 +1,304 @@
+control plane's API server component remains available.
+
+<!-- body -->
+
+## Language overview
+
+The [CEL
+language](https://github.com/google/cel-spec/blob/master/doc/langdef.md) has a
+straightforward syntax that is similar to the expressions in with C, C++, Java,
+JavaScript and Go.
+
+CEL was designed to be embedded into applications. Each CEL "program" is a
+single expression that evaluates to a single value. CEL expressions are
+typically short "one-liners" that inline well into the string fields of Kubernetes
+API resources.
+
+Inputs to a CEL program are "variables". Each Kubernetes API field that contains
+CEL declares in the API documentation which variables are available to use for
+that field. For example, in the `x-kubernetes-validations[i].rules` field of
+CustomResourceDefinitions, the `self` and `oldSelf` variables are available and
+refer to the previous and current state of the custom resource data to be
+validated by the CEL expression. Other Kubernetes API fields may declare
+different variables. See the API documentation of the API fields to learn which
+variables are available for that field.
+
+Example CEL expressions:
+
+{{< table caption="Examples of CEL expressions and the purpose of each" >}}
+| Rule                                                                               | Purpose                                                                           |
+|------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------|
+| `self.minReplicas <= self.replicas && self.replicas <= self.maxReplicas`           | Validate that the three fields defining replicas are ordered appropriately        |
+| `'Available' in self.stateCounts`                                                  | Validate that an entry with the 'Available' key exists in a map                   |
+| `(self.list1.size() == 0) != (self.list2.size() == 0)`                             | Validate that one of two lists is non-empty, but not both                         |
+| `self.envars.filter(e, e.name = 'MY_ENV').all(e, e.value.matches('^[a-zA-Z]*$')`   | Validate the 'value' field of a listMap entry where key field 'name' is 'MY_ENV'  |
+| `has(self.expired) && self.created + self.ttl < self.expired`                      | Validate that 'expired' date is after a 'create' date plus a 'ttl' duration       |
+| `self.health.startsWith('ok')`                                                     | Validate a 'health' string field has the prefix 'ok'                              |
+| `self.widgets.exists(w, w.key == 'x' && w.foo < 10)`                               | Validate that the 'foo' property of a listMap item with a key 'x' is less than 10 |
+| `type(self) == string ? self == '100%' : self == 1000`                             | Validate an int-or-string field for both the the int and string cases             |
```

Comment:
Just curiously, why is the in value 1000 instead of 100 ?

Reply:
I think we could use a different value to make it extra clear the numbers aren't related. Eg, check either for `"100%"` or `7`, reject any other value.

Reply:
I can use random numbers here. I think the use cases I copied this from was implicitly using millis as the unit of the int, but I agree that this example raises eyebrows the way it is currently written. Fixed.